src/extensions/replay/sessionrecording.ts

```diff
@@ -343,13 +361,37 @@ export class SessionRecording {
             return 'buffering'
         }
 
+        if (this.urlTriggerStatus === 'pending') {
+            return 'buffering'
+        }
+
         if (isBoolean(this.isSampled)) {
             return this.isSampled ? 'sampled' : 'disabled'
         } else {
             return 'active'
         }
     }
 
+    private get urlTriggerStatus(): 'activated' | 'pending' | 'disabled' {
+        if (this.receivedDecide && this._urlTriggers.length === 0) {
+            return 'disabled'
+        }
+
+        const currentValue = this.instance?.get_property(SESSION_RECORDING_URL_TRIGGER_ACTIVATED)
+
+        if (TRIGGER_STATUSES.includes(currentValue)) {
+            return currentValue as TriggerStatus
+        }
+
+        return 'pending'
+    }
+
+    private set urlTriggerStatus(status: TriggerStatus) {
+        this.instance?.persistence?.register({
```

Comment:
both instance, and persistence are optional according to the types. Is it possible we don't have them set, should I add a fallback to a class property for that case?

Reply:
i keep meaning to test this... I _think_ that persistence will always at least be memory and so even though technically we might not have an instance and it might not have persistence we will always have them if recording is running

we do local flag fallback elsewhere to protect us but I _think_ it's actually unnecessary since I believe we always have memory persistence

(i've just never made time to check if that's true and simplify if so)